packages/cli/src/WorkflowExecuteAdditionalData.ts

```diff
@@ -508,36 +509,25 @@ function hookFunctionsSave(parentProcessMode?: string): IWorkflowExecuteHooks {
 						}
 					}
 
-					const workflowSettings = this.workflowData.settings;
-					let saveManualExecutions = config.getEnv('executions.saveDataManualExecutions');
-					if (workflowSettings?.saveManualExecutions !== undefined) {
-						// Apply to workflow override
-						saveManualExecutions = workflowSettings.saveManualExecutions as boolean;
-					}
+					const defaults = {
+						saveDataErrorExecution: config.getEnv('executions.saveDataOnError'),
+						saveDataSuccessExecution: config.getEnv('executions.saveDataOnSuccess'),
+						saveManualExecutions: config.getEnv('executions.saveDataManualExecutions'),
+					};
```

Comment:
Instead of setting up this object here, causing all calls to `toSaveSettings` to re-create the same code, we might want to move this to inside the function, reducing duplication in future usage wdyt?